bim2sim/tasks/bps/plot_results.py

```diff
@@ -270,86 +281,153 @@ def save_or_show_plot(save_path, dpi, format='pdf'):
+            area_specific (bool): True if result_str values should be divided
+             by area to get valuer per square meter.
+
+        TODO: Aggregated Zones
+        Combined zones, how to proceed:
+            - All rooms in the combined zone are given the same color and
+                the same value
+            - Rooms need names in the plot
+            - Legend in the margin showing which room name belongs to which
+                zone
+
+
+            Generally revise:
+            - Unit in the color mapping plot and in the plot for numerical
+                values
         """
-
+        # TODO this is currently not working for aggregated zones.
+        # check if result_str is valid for floor plan visualization
+        if result_str not in self.playground.sim_settings.sim_results:
+            raise ValueError(f'Result {result_str} was not requested by '
+                             f'sim_setting "sim_results" or is not provided'
+                             f'by the simulation. '
+                             f'Please Check your "sim_results" settings.')
+        if "_rooms" not in result_str:
+            raise ValueError(f'Result {result_str} does not provide room level'
+                             f'information. Floor plan visualization is  only '
+                             f'available for room level results.')
         # create the dict with all space guids and resulting values in the
         # first run
-        # TODO this is currently not working for aggregated zones.
         svg_adjust_dict = {}
         for col_name, col_data in df.items():
-            if 'heat_demand_rooms_' in col_name and 'total' not in col_name:
-                space_guid = col_name.split('heat_demand_rooms_')[-1]
-                storey_guid = \
-                    self.playground.state['tz_elements'][space_guid].storeys[
-                        0].guid
+            if result_str + '_' in col_name and 'total' not in col_name:
+                space_guid = col_name.split(result_str + '_')[-1]
+                storey_guid = None
+                space_area = None
+                for guid, ele in elements.items():
+                    if guid == space_guid:
+                        # TODO use all storeys for aggregated zones
+                        if isinstance(ele, SerializedElement):
+                            storey_guid = ele.storeys[0]
+                        else:
+                            storey_guid = ele.storeys[0].guid
+                        space_area = ele.net_area
+
+                if not storey_guid or not space_area:
+                    self.logger.warning(
+                        f"For space with guid {space_guid} no"
+                        f" fitting storey could be found. This space will be "
+                        f"ignored for floor plan plots. ")
+                    continue
+                # Ignore very small areas
+                min_area = 2 * ureg.m ** 2
```

Comment:
add this as a default variable to the setting of the function. If we have a small house with tiny rooms, we may need a limit smaller than 2m²

Reply:
done